### ISSUE TITLE
Abnormal cpu and memory usage

### ISSUE THREAD

**Issue by Mek101** [linuxmint/Cinnamon#7121]
```
 * Cinnamon version 3.6.6
 * Distribution - From Linux Mint 18 to 18.3
 * Nvidia GTX 950 *and* proprietary driver nvidia-387
 * 64 bit
 ```

**Issue**
Cinnamon eats up to 25% cpu (100% of one core) at random spikes, making the GUI freeze, the more time passes, the more those freezes become longer and more frequent, until the desktop becomes unusable. To only way to temporarily solve the problem is restarting Cinnamon every 4-5 minutes. I've already tried different things:
Updating the kernel
Updating the video driver
Disabling effects
Renaiming $HOME/.cinnamon
Useless to say that none of them worked. I've also already asked about this issue in the mint forums here: 
https://forums.linuxmint.com/viewtopic.php?f=208&t=254604&p=1371857#p1371857
But it keeps to reappear again and again.
It also produces tearing while moving windows, during the visual effects, while watching videos (both youtube and local ones) and playing games, but the main problem is still the randomic freezing

**Other information**
```
mek101@mint18 ~ $ inxi -Fxz
System:    Host: mint18 Kernel: 4.11.0-14-generic x86_64 (64 bit gcc: 5.4.0)
           Desktop: Cinnamon 3.6.6 (Gtk 3.18.9-1ubuntu3.3)
           Distro: Linux Mint 18.3 Sylvia
Machine:   Mobo: ASRock model: FM2A88M-HD+
           Bios: American Megatrends v: P3.10 date: 05/05/2015
CPU:       Quad core AMD A10-6800K APU with Radeon HD Graphics (-MCP-) cache: 8192 KB
           flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm) bmips: 37533
           clock speeds: max: 4691 MHz 1: 4691 MHz 2: 4691 MHz 3: 4691 MHz
           4: 4691 MHz
Graphics:  Card-1: Advanced Micro Devices [AMD/ATI] Richland [Radeon HD 8670D]
           bus-ID: 00:01.0
           Card-2: NVIDIA GM206 [GeForce GTX 950] bus-ID: 01:00.0
           Display Server: X.Org 1.18.4 drivers: nvidia,ati,radeon (unloaded: fbdev,vesa,nouveau)
           Resolution: 1920x1080@60.00hz
           GLX Renderer: GeForce GTX 950/PCIe/SSE2
           GLX Version: 4.6.0 NVIDIA 387.34 Direct Rendering: Yes
Audio:     Card-1 Advanced Micro Devices [AMD] FCH Azalia Controller
           driver: snd_hda_intel bus-ID: 00:14.2
           Card-2 NVIDIA Device 0fba driver: snd_hda_intel bus-ID: 01:00.1
           Card-3 Advanced Micro Devices [AMD/ATI] Trinity HDMI Audio Controller
           driver: snd_hda_intel bus-ID: 00:01.1
           Card-4 Microsoft LifeChat LX-3000 Headset
           driver: USB Audio usb-ID: 004-002
           Sound: Advanced Linux Sound Architecture v: k4.11.0-14-generic
Network:   Card: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller
           driver: r8169 v: 2.3LK-NAPI port: d000 bus-ID: 02:00.0
           IF: enp2s0 state: up speed: 1000 Mbps duplex: full mac: <filter>
Drives:    HDD Total Size: 1500.3GB (33.0% used)
           ID-1: /dev/sda model: Samsung_SSD_840 size: 500.1GB
           ID-2: /dev/sdb model: WDC_WD10EZEX size: 1000.2GB
Partition: ID-1: / size: 332G used: 76G (25%) fs: ext4 dev: /dev/sda5
           ID-2: swap-1 size: 0.26GB used: 0.00GB (0%) fs: swap dev: /dev/sda6
RAID:      No RAID devices: /proc/mdstat, md_mod kernel module present
Sensors:   System Temperatures: cpu: 0.0C mobo: 32.0C gpu: 0.0:51C
           Fan Speeds (in rpm): cpu: N/A fan-1: 3461 fan-2: 1222 fan-3: 0 fan-4: 0 fan-5: 0
Info:      Processes: 236 Uptime: 23 min Memory: 2202.7/7885.0MB
           Init: systemd runlevel: 5 Gcc sys: 5.4.0
           Client: Shell (bash 4.3.481) inxi: 2.2.35
```

**Comment by Mek101** [comment]
Update: seems like some sort of memory leak.
I noticed `cinnamon --replace` grows in memory usage with time. I closed it once it reached 300mb, and after restarting it was only 95mb.
So yeah, i think there's some sort of memory leak.

**Comment by pizzadude** [comment]
Do you have indicator support enabled? If so disable it.

**Comment by zephred** [comment]
I am having the same problem with the same version of cinnamon (3.6.6) on an up-to-date ubuntu 16.04. The RES  memory usage went once up to more than 10GB... 
Re-starting cinnamon reduces its RES memory usage down to ~200mb. 
Indicator support is already disabled.

```
$ inxi -Fxz
System:    Host: sky Kernel: 4.4.0-103-generic x86_64 (64 bit gcc: 5.4.0) Desktop: Cinnamon 3.6.6 (Gtk 2.24.30)
           Distro: Ubuntu 16.04 xenial
Machine:   System: Dell (portable) product: XPS 13 9360
           Mobo: Dell model: 06CC14 v: A00 Bios: Dell v: 2.1.0 date: 08/02/2017
CPU:       Dual core Intel Core i7-7500U (-HT-MCP-) cache: 4096 KB
           flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx) bmips: 11615
           clock speeds: max: 3500 MHz 1: 719 MHz 2: 692 MHz 3: 705 MHz 4: 672 MHz
Graphics:  Card: Intel Device 5916 bus-ID: 00:02.0
           Display Server: X.Org 1.18.4 driver: N/A Resolution: 3200x1800@59.98hz
           GLX Renderer: Mesa DRI Intel HD Graphics 620 (Kabylake GT2)
           GLX Version: 3.0 Mesa 17.0.7 Direct Rendering: Yes
Audio:     Card Intel Device 9d71 driver: snd_hda_intel bus-ID: 00:1f.3 Sound: ALSA v: k4.4.0-103-generic
Network:   Card: Qualcomm Atheros QCA6174 802.11ac Wireless Network Adapter driver: ath10k_pci bus-ID: 3a:00.0
           IF: wlp58s0 state: up mac: <filter>
Drives:    HDD Total Size: NA (-) ID-1: /dev/nvme0n1 model: N/A size: 1024.2GB
Partition: ID-1: / size: 904G used: 394G (46%) fs: ext4 dev: /dev/nvme0n1p3
           ID-2: swap-1 size: 34.12GB used: 0.12GB (0%) fs: swap dev: /dev/nvme0n1p4
RAID:      No RAID devices: /proc/mdstat, md_mod kernel module present
Sensors:   System Temperatures: cpu: 63.0C mobo: N/A
           Fan Speeds (in rpm): cpu: N/A
Info:      Processes: 292 Uptime: 2 days Memory: 8571.9/15932.2MB Init: systemd runlevel: 5 Gcc sys: 5.4.0
           Client: Shell (bash 4.3.481) inxi: 2.2.35 
```

**Comment by PaulVD** [comment]
Same issue, on one machine only - the other has never been affected. I keep switching back to Mate (which does not have the problem) when the Cinnamon desktop becomes unusable. But I like Cinnamon better, so every time a new version of Mint comes out I try again. I have had the problem from Mint 17.3, 18.1, 18.2 and now 18.3, with various kernels.
After a clean install it takes perhaps a couple of weeks before the issue arises, and as time goes on it gets worse and worse. Restarting Cinnamon with Ctrl-Alt-Esc fixes the problem for a few minutes, but it soon comes back. Rebooting also gives only a few minutes respite. So this is a very peculiar problem - there is clearly some configuration data that Cinnamon is saving that does not get reset on a reboot and that gets progressively worse.
There is ample memory and disk space, on an 8-processor Core i7.

```
$ inxi -Fxz
System:    Host: Nimrod Kernel: 4.10.0-40-generic x86_64 (64 bit gcc: 5.4.0)
           Desktop: Cinnamon 3.6.6 (Gtk 3.18.9-1ubuntu3.3) Distro: Linux Mint 18.3 Sylvia
Machine:   Mobo: ASUSTeK model: H170M-PLUS v: Rev X.0x Bios: American Megatrends v: 3404 date: 07/10/2017
CPU:       Quad core Intel Core i7-6700K (-HT-MCP-) cache: 8192 KB
           flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx) bmips: 32064
           clock speeds: max: 4200 MHz 1: 1168 MHz 2: 3899 MHz 3: 3865 MHz 4: 917 MHz 5: 2097 MHz 6: 3507 MHz
           7: 3799 MHz 8: 1166 MHz
Graphics:  Card: Intel Sky Lake Integrated Graphics bus-ID: 00:02.0
           Display Server: X.Org 1.18.4 drivers: intel (unloaded: fbdev,vesa)
           Resolution: 1024x768@60.00hz, 1920x1200@59.95hz
           GLX Renderer: Mesa DRI Intel HD Graphics 530 (Skylake GT2)
           GLX Version: 3.0 Mesa 17.0.7 Direct Rendering: Yes
Audio:     Card Intel Sunrise Point-H HD Audio driver: snd_hda_intel bus-ID: 00:1f.3
           Sound: Advanced Linux Sound Architecture v: k4.10.0-40-generic
Network:   Card: Intel Ethernet Connection (2) I219-V driver: e1000e v: 3.2.6-k bus-ID: 00:1f.6
           IF: enp0s31f6 state: up speed: 100 Mbps duplex: full mac: <filter>
Drives:    HDD Total Size: 1000.2GB (15.8% used) ID-1: /dev/sda model: WDC_WD10J31X size: 1000.2GB
Partition: ID-1: / size: 20G used: 9.5G (53%) fs: ext4 dev: /dev/sda7
           ID-2: /home size: 289G used: 124G (46%) fs: ext4 dev: /dev/sda4
           ID-3: swap-1 size: 15.73GB used: 0.00GB (0%) fs: swap dev: /dev/dm-0
RAID:      No RAID devices: /proc/mdstat, md_mod kernel module present
Sensors:   System Temperatures: cpu: 29.8C mobo: 27.8C
           Fan Speeds (in rpm): cpu: 0
Info:      Processes: 262 Uptime: 5:08 Memory: 1431.0/15900.4MB Init: systemd runlevel: 5 Gcc sys: 5.4.0
           Client: Shell (bash 4.3.481) inxi: 2.2.35 
```

**Comment by IlyukhinAndrey** [comment]
The same issue 

The size of memory occupied by the cinnamon- process increasing. And after 8 hours, there is already more 1 gigabyte.
<script src="https://gist.github.com/anonymous/00c5606102b8e057370bb6152ad258c3.js"></script>

**Comment by IlyukhinAndrey** [comment]
-cinnamon 800Mb Two hour after restart.
![2017-12-14 12-13-29](https://user-images.githubusercontent.com/34511109/33984439-a8d8290e-e0c8-11e7-88ac-6afa5a8ca630.png)

**Comment by jaszhix** [comment]
For those that are listing system specs, can you guys run this command? It will be more more helpful for this issue.

Run `gsettings get org.cinnamon enabled-applets && gsettings get org.cinnamon enabled-desklets && gsettings get org.cinnamon enabled-extensions`. It will list all the xlets that are currently in use. Third party xlets can do anything Cinnamon can, and is a factor that should always be known when reporting memory/cpu problems.

**Comment by PaulVD** [comment]
paul@Nimrod ~ $ gsettings get org.cinnamon enabled-applets
['panel1:right:0:systray@cinnamon.org:0', 'panel1:left:0:menu@cinnamon.org:1', 'panel1:left:2:panel-launchers@cinnamon.org:3', 'panel1:left:3:window-list@cinnamon.org:4', 'panel1:right:1:keyboard@cinnamon.org:5', 'panel1:right:2:notifications@cinnamon.org:6', 'panel1:right:3:removable-drives@cinnamon.org:7', 'panel1:right:4:user@cinnamon.org:8', 'panel1:right:5:network@cinnamon.org:9', 'panel1:right:6:bluetooth@cinnamon.org:10', 'panel1:right:7:power@cinnamon.org:11', 'panel1:right:8:calendar@cinnamon.org:12', 'panel1:right:9:sound@cinnamon.org:13']
paul@Nimrod ~ $ gsettings get org.cinnamon enabled-desklets
@as []
paul@Nimrod ~ $ gsettings get org.cinnamon enabled-extensions
@as []

Github would not let me upload the requested files directly, but here are copies.
[xsession-errors.txt](https://github.com/linuxmint/Cinnamon/files/1558775/xsession-errors.txt)
[cinnamon.glass.log](https://github.com/linuxmint/Cinnamon/files/1558778/cinnamon.glass.log)

**Comment by IlyukhinAndrey** [comment]
After 6 hours it is 1,4 Gb in the process list.
![2017-12-14 16-00-23](https://user-images.githubusercontent.com/34511109/33993647-72239842-e0e8-11e7-8781-fafdac7e4c4e.png)

**Comment by jaszhix** [comment]
@IlyukhinAndrey You didn't provide any info about your Cinnamon environment, like the Cinnamon version, xlets used, log data I mentioned above, etc.

**Comment by IlyukhinAndrey** [comment]
andrey@Andrey ~ $ gsettings get org.cinnamon enabled-applets && gsettings get org.cinnamon enabled-desklets && gsettings get org.cinnamon enabled-extensions
['panel1:left:0:menu@cinnamon.org:0', 'panel1:left:1:show-desktop@cinnamon.org:1', 'panel1:left:2:panel-launchers@cinnamon.org:2', 'panel1:left:3:window-list@cinnamon.org:3', 'panel1:right:0:notifications@cinnamon.org:4', 'panel1:right:4:user@cinnamon.org:5', 'panel1:right:5:removable-drives@cinnamon.org:6', 'panel1:right:6:keyboard@cinnamon.org:7', 'panel1:right:7:network@cinnamon.org:9', 'panel1:right:8:sound@cinnamon.org:10', 'panel1:right:9:power@cinnamon.org:11', 'panel1:right:10:systray@cinnamon.org:12', 'panel1:right:11:calendar@cinnamon.org:13', 'panel1:right:12:windows-quick-list@cinnamon.org:14', 'panel1:right:1:gputemperature@silentage.com:16', 'panel1:right:2:temperature@fevimu:17']
@as []
@as []
[glass.log](https://github.com/linuxmint/Cinnamon/files/1559318/glass.log)


[xsession-errors.txt](https://github.com/linuxmint/Cinnamon/files/1559322/xsession-errors.txt)

**Comment by Mek101** [comment]
@jaszhix 
```
mek101@mint18 ~ $ gsettings get org.cinnamon enabled-applets && gsettings get org.cinnamon enabled-desklets && gsettings get org.cinnamon enabled-extensions
['panel1:left:0:menu@cinnamon.org:1', 'panel1:center:0:window-list@cinnamon.org:4', 'panel1:right:1:systray@cinnamon.org:0', 'panel1:right:0:notifications@cinnamon.org:6', 'panel1:right:4:removable-drives@cinnamon.org:7', 'panel1:right:9:user@cinnamon.org:8', 'panel1:right:5:network@cinnamon.org:9', 'panel1:right:8:calendar@cinnamon.org:12', 'panel1:right:7:sound@cinnamon.org:13']
@as []
@as []
mek101@mint18 ~ $ 
```
[glass.log and .xsession-errors](https://mega.nz/#F!UBIAQawK!KE8aO1HoULApWL2mRC917w)

@pizzadude 
Indicator support now disabled. Let's see if it works.

**Comment by zephred** [comment]
Here is my output of gsettings:
`$gsettings get org.cinnamon enabled-applets && gsettings get org.cinnamon enabled-desklets && gsettings get org.cinnamon enabled-extensions
['panel1:left:0:menu@cinnamon.org:0',
 'panel1:left:1:show-desktop@cinnamon.org:1',
 'panel1:left:2:panel-launchers@cinnamon.org:2',
 'panel1:left:3:window-list@cinnamon.org:3',
 'panel1:right:7:notifications@cinnamon.org:4',
 'panel1:right:11:user@cinnamon.org:5',
 'panel1:right:12:removable-drives@cinnamon.org:6',
 'panel1:right:14:network@cinnamon.org:9',
 'panel1:right:15:sound@cinnamon.org:10',
 'panel1:right:16:power@cinnamon.org:11',
 'panel1:right:18:systray@cinnamon.org:12',
 'panel1:right:19:calendar@cinnamon.org:13',
 'panel1:right:20:windows-quick-list@cinnamon.org:14',
 'panel1:right:8:temperature@fevimu:16',
 'panel1:right:9:weather@mockturtl:31',
 'panel1:right:10:cpufreq@mtwebster:32',
 'panel1:right:6:sysmonitor@orcus:34',
 'panel1:right:17:blueberry@cinnamon.org:35',
 'panel1:right:1:keyboard@cinnamon.org:39']
@as []
@as [] `

The measured growth of RES mem usage is roughly 1kB/minute.

**Comment by astephanh** [comment]
Hi, 

i too have about 3.3GB Ram usage after 2 days.

[glass.log.txt](https://github.com/linuxmint/Cinnamon/files/1565927/glass.log.txt)
[xsession-errors.txt](https://github.com/linuxmint/Cinnamon/files/1565928/xsession-errors.txt)


melvin@r2d2 ~
└─ 07:29 PM $ gsettings get org.cinnamon enabled-applets ; echo;  gsettings get org.cinnamon enabled-desklets ; echo ;gsettings get org.cinnamon enabled-extensions
['panel1:left:0:menu@cinnamon.org:0', 'panel1:left:1:show-desktop@cinnamon.org:1', 'panel1:left:2:panel-launchers@cinnamon.org:2', 'panel1:left:3:window-list@cinnamon.org:3', 'panel1:right:0:notifications@cinnamon.org:4', 'panel1:right:3:keyboard@cinnamon.org:7', 'panel1:right:4:bluetooth@cinnamon.org:8', 'panel1:right:5:network@cinnamon.org:9', 'panel1:right:6:sound@cinnamon.org:10', 'panel1:right:8:systray@cinnamon.org:12', 'panel1:right:9:calendar@cinnamon.org:13', 'panel1:right:10:windows-quick-list@cinnamon.org:14', 'panel1:right:0:multicore-sys-monitor@ccadeptic23:15', 'panel1:right:0:temperature@fevimu:17']

@as []

**Comment by IlyukhinAndrey** [comment]
I didn't power off my computer on Friday just for interest. It should be interesting to see on the size of process on Mondays morning :)

**Comment by Mek101** [comment]
@IlyukhinAndrey good luck then

Does some know hot to tag this thread as "bug"?

**Comment by leigh123linux** [comment]
@Mek101  Why would we tag it?, no one has provided proof it's cinnamon that's leaking.
The only common denominator I can see is the crappy xenial base and some ancient Xorg version :-)

**Comment by leigh123linux** [comment]
Closed

Duplicate of #6850

**Comment by IlyukhinAndrey** [comment]
@Mek101  About 4,7Gb ))) I solved the issue. The reason were the GPU and CPU temperature applets in status bar. I didn`t try to define witch one is bad, but when I turned it off and restart -cinnamon it was finished to increase in memory. So if any body have such problem, try to turn OFF, it helped in my case: gputemperature@silentage.com or/and temperature@fevimu, or something like that...
![2017-12-18 09-44-52](https://user-images.githubusercontent.com/34511109/34097984-fd1e3104-e3eb-11e7-8453-254c01d28115.png)
